drift-idl-generator/src/main/java/com/facebook/drift/idl/generator/ThriftIdlGenerator.java

```diff
@@ -129,6 +131,46 @@ public ThriftIdlGenerator(ThriftIdlGeneratorConfig config, ClassLoader classLoad
 
         this.namespaces = config.getNamespaces();
         this.recursive = config.isRecursive();
+
+        for (String customCodecClassName : config.getCustomCodecs()) {
+            try {
+                Class<?> codecClass = load(customCodecClassName);
+                if (codecClass == null) {
+                    continue;
+                }
+                Object codec;
+
+                try {
+                    codec = codecClass.getDeclaredConstructor(ThriftCatalog.class).newInstance(catalog);
+                }
+                catch (NoSuchMethodException e) {
+                    try {
+                        codec = codecClass.getDeclaredConstructor().newInstance();
+                    }
+                    catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException nestedException) {
+                        throw new ThriftIdlGeneratorException("Can not find constructor for the custom codec.");
+                    }
+                }
+
+                if (codec instanceof ThriftCodec) {
+                    ThriftCodec<?> thriftCodec = (ThriftCodec<?>) codec;
+                    if (thriftCodec.getType() != null) {
```

Comment:
IMO The codec has to provide type information statically.  

When the `getType` method is not static it can in theory return different ThriftType based on how it is initialized. Think about the JSON wrapper codecs you implemented. Based on a configuration the codec may chose a different name for a wrapper structure, or represent JSON field either as string or binary.

To avoid breaking an interface I would suggest adding a special annotation, for example `@CodecThriftType` and expect a codec implementation to expose a static, zero argument method annotated with this annotation returning a `ThriftType`. Then make only codec exposing ThriftType statically supported by the IDL plugin.